### PR TITLE
Adds ADMIN ONLY  pocket dimensions

### PR DESCRIPTION
**Pull request by Djiq**
# About The Pull Request

Adds basic support for pocket dimensions, allowing admins to easily create a new z level, add a room in it, attach events to it if it breaks(there aren't many currently) and spawn portals in it, all with simple TGUI to make the whole job easier.

Job commisioned by Stickymayhem

## Why It's Good For The Game

More cool admin tools are always amazing to have.

## Changelog
:cl:
admin: added admin only pocket dimensions that allow admemes to make events easily or test random things in them.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
why not make this part of the reserved z level which is basically just for stuff like this

**Comment by EdgeLordExe** [comment]
> why not make this part of the reserved z level which is basically just for stuff like this

it is a reserved z level, but due to how customizable it is, it would clash with other things that could be on the reserved z level, so better to just keep it in it's own z level.

**Comment by ATH1909** [comment]
Rope Trick (or Create Demiplane) spell for wizards when???

File: code/game/turfs/open/chasm.dm [hunk 1]
```diff
@@ -112,3 +116,26 @@
...
+	var/list/numlist = list()
+	zlist -= SSmapping.levels_by_trait(ZTRAIT_CENTCOM)
+	zlist -= SSmapping.levels_by_trait(ZTRAIT_POCKETDIM)
+	for(var/X in zlist)
+		var/datum/space_level/level = X
+		numlist += level.z_value
+	return locate(loc.x,loc.y,pick(numlist))
+
+/turf/open/chasm/void/lavaland/chasm_destination()
+	///I know this isnt always lavaland but shut the fuck up ok?
```
Comment (SpaceManiac) [inline code/game/turfs/open/chasm.dm:139]: it would clash with other things that could be on the reserved z level, so better to just keep it in it's own z level.

**Comment by FloranOtten** [comment]
Damn, there goes the impressiveness of my create z level commands.

**Comment by SpaceManiac** [comment]
> it would clash with other things that could be on the reserved z level, so better to just keep it in it's own z level.

Clash how? Isolated temporary rooms is the explicit purpose of the existing turf reservation system.

**Comment by optimumtact** [comment]
Also I'm not sure we can afford the extra memory if we do give it it's own z level

File: tgui/packages/tgui/interfaces/PocketDimensionCustomizer.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        <Box align="center" my={1}>
+          <Box >
+            Dimension Controls
+          </Box>
+
+          <Box>
+            <Button
+              content="Create Dimension"
+              onClick={() => act('create_dimension')}
+              tooltip={`DO THIS FIRST, causes some lag, may \
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/PocketDimensionCustomizer.js:23]: 1. With backticks, you do not need to escape newlines with backslashes.
2. Use the multiline string tag, because it removes all unnecessary whitespace from the string at compile time, and is better for the bundle size. For it to work, you must treat it like a separate block of text.

Example:
```js
multiline`
  Foo Bar
  Hippity Hoppity
`
```

How it's stored in the bundle:

```js
"Foo Bar/nHippity Hoppity"
```

**Comment by EdgeLordExe** [comment]
> Also I'm not sure we can afford the extra memory if we do give it it's own z level

the z level isnt creates round start but when admin needs it, effectively throwing off the guilt to the admin that well knew it could crash the server.
